File: app/src/main/java/org/ashwin/opencut/presentation/preview/VideoPreviewView.java
package org.ashwin.opencut.presentation.preview;

import android.content.Context;
import android.net.Uri;
import android.opengl.GLSurfaceView;

import org.ashwin.opencut.data.media.PreviewRenderer;
import org.ashwin.opencut.domain.model.EffectSettings;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class VideoPreviewView extends GLSurfaceView {

    private final PreviewRenderer renderer;

    public VideoPreviewView(Context context) {
        super(context);

        setEGLContextClientVersion(2);

        renderer = new PreviewRenderer(context);

        setRenderer(renderer);

        setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY);
    }

    public void loadVideo(@Nullable Uri uri) {
        renderer.loadVideo(uri);
    }

    public void setEffect(EffectSettings effectSettings) {
        renderer.setEffect(effectSettings);
    }
}

File: app/src/main/java/org/ashwin/opencut/presentation/preview/VideoPreviewView.kt
package org.ashwin.opencut.presentation.preview

import android.content.Context
import android.net.Uri
import android.opengl.GLSurfaceView
import org.ashwin.opencut.data.media.PreviewRenderer
import org.ashwin.opencut.domain.model.EffectSettings

class VideoPreviewView(context: Context) : GLSurfaceView(context) {

    private val renderer: PreviewRenderer = PreviewRenderer(context)

    init {
        setEGLContextClientVersion(2)
        setRenderer(renderer)
        renderMode = RENDERMODE_CONTINUOUSLY
    }

    fun loadVideo(uri: Uri?) {
        renderer.loadVideo(uri)
    }

    fun setEffect(effectSettings: EffectSettings) {
        renderer.setEffect(effectSettings)
    }

    fun togglePlayback() {
        renderer.togglePlayback()
    }

    fun isPlaying(): Boolean {
        return renderer.isPlaying
    }
}
